tsdb/compact.go

```diff
@@ -196,6 +206,10 @@ func NewLeveledCompactorWithOptions(ctx context.Context, r prometheus.Registerer
 	if pe == nil {
 		pe = index.EncodePostingsRaw
 	}
+	pd := opts.PD
+	if pd == nil {
+		pd = DefaultPostingsDecoderFactory
```

Comment:
There are three places where `nil` is reinterpreted as a default.  Do we need three?